### ISSUE TITLE
Can not adopt Olimex POE BT proxy in ESPhome

### ISSUE THREAD

**Issue by Mariusthvdb** [esphome/dashboard#320]
### The problem

because it is LAN connected, and not Wifi.

The device is recognized correctly in the Integration panel after having flashed the firmware successfully via https://esphome.github.io/bluetooth-proxies/

Note there is no option offered in that Next box to Include in Home Assistant, as is shown in the release party for the Wifi M% stack.

Ive had a long an intense discussion on Discord, and an [active topic in the community](https://community.home-assistant.io/t/how-to-adopt-the-olimex-poe-board-without-wifi-credentials/460871), neither of which have yet solved the issue, hence I have to report here as true issue:

summary: can not adopt POE device because of Wifi only adapting in ESPHome dashboard.

TLDR:
upon opening ESPhome dashboard, the device is 'discovered'

<img width="1021" alt="Schermafbeelding 2022-09-13 om 20 52 33" src="https://user-images.githubusercontent.com/33354141/190401384-ecb75b7e-a69e-404a-a4b7-9ac7d2db492a.png">

clicking adopt wont work, because it wants wifi credentials. However, upon doing so a second instance is displayed:

<img width="991" alt="Schermafbeelding 2022-09-13 om 20 52 43" src="https://user-images.githubusercontent.com/33354141/190401559-c66d472a-ca63-4a4d-954c-a86bda6edf43.png">

directly followed by an offline of the same device:
<img width="992" alt="Schermafbeelding 2022-09-13 om 20 52 53" src="https://user-images.githubusercontent.com/33354141/190401658-c69eafe1-becf-4bb8-88fe-77c89be42366.png">

we can Edit the yamlfiles, add the configuration as provided at https://github.com/esphome/bluetooth-proxies/blob/main/olimex-esp32-poe-iso.yaml which initially seems to succeed, a connection is made.

however, closing the log window reveals:
<img width="961" alt="Schermafbeelding 2022-09-13 om 19 26 10" src="https://user-images.githubusercontent.com/33354141/190401944-ee3f1562-7174-4c12-b4ed-73dce6b7effa.png">

there seems to be no way to get out of this circle.

also tried it with static IP, but the result is the same. running in circles.

### What version of Home Assistant Core has the issue?

2022.9.4

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

ESPhome

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/esphome/

### Example YAML snippet

```yaml
substitutions:
  name: olimex-bluetooth-proxy

esphome:
  name: ${name}
  name_add_mac_suffix: true
  project:
    name: esphome.bluetooth-proxy-1
    version: "1.0"

esp32:
  board: esp32-poe-iso
  framework:
    type: arduino

ethernet:
  type: LAN8720
  mdc_pin: GPIO23
  mdio_pin: GPIO18
  clk_mode: GPIO17_OUT
  phy_addr: 0
  power_pin: GPIO12
  manual_ip:
    static_ip: 192.168.1.135
    gateway: 192.168.1.1
    subnet: 255.255.255.0

api:
logger:
ota:

dashboard_import:
  package_import_url: github://esphome/bluetooth-proxies/olimex-esp32-poe-iso.yaml@main

esp32_ble_tracker:
  scan_parameters:
    interval: 1100ms
    window: 1100ms
    active: true

bluetooth_proxy:

button:
- platform: safe_mode
  name: Safe Mode Boot
  entity_category: diagnostic
```

**Comment by jesserockz** [comment]
@balloob The adoption process needs to be able to know the device is set up for ethernet, not wifi and skip the wifi step

**Comment by Mariusthvdb** [comment]
there's a lot of extra non-ESPHome dashboard posts in that Community post, so I summarized it some more in the bottom post https://community.home-assistant.io/t/how-to-adopt-the-olimex-poe-board-without-wifi-credentials/460871/42?u=mariusthvdb

**Comment by Mariusthvdb** [comment]
before updating to the soon to be released .10 add-on, and making the mistake of trying to include these POE proxies, are we safe to go? Must  confess I am somewhat confused about the developments on the matter ;-)

**Comment by Mariusthvdb** [comment]
thx for the fix. I can adopt it now. After a refresh of the board on https://esphome.github.io/bluetooth-proxies/ 
 I can see the devices being discovered:
<img width="1471" alt="Schermafbeelding 2022-10-07 om 10 54 06" src="https://user-images.githubusercontent.com/33354141/194514740-c4621234-5fd5-4ffc-9a56-ee7b461e555a.png">

so, I click on Adopt, change the name to something nicer and that seems to fare well. I do notice it still compiles Wifi related lines, and an error scrolls by:

```
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-spi.c: In function 'spiTransferBytesNL':
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-spi.c:922:39: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
                 uint8_t * last_out8 = &result[c_longs-1];
                                       ^
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-spi.c:923:40: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
                 uint8_t * last_data8 = &last_data;
                                        ^
```
no matte that, there is success reported, and the board is online. hurray.

however, in the dashboard the discovered device is still showing:

**Comment by Mariusthvdb** [comment]
deleted reply and reposted in https://github.com/esphome/dashboard/issues/329
since this is now closed
